analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -187,13 +230,14 @@ public override void ExecutionCompleted()
     private ProgramState ProcessInvocation(SymbolicContext context, IInvocationOperationWrapper invocation)
     {
         var targetMethod = invocation.TargetMethod;
-        if (targetMethod.Is(KnownType.System_Linq_Enumerable, nameof(Enumerable.Count))
-            && SizeConstraint(context.State, invocation.Instance ?? invocation.Arguments[0].ToArgument().Value, HasFilteringPredicate()) is { } constraint)
-        {
-            return context.SetOperationConstraint(constraint);
-        }
-        else if (invocation.Instance is { } instance)
+        if ((invocation.Instance ?? invocation.Arguments.FirstOrDefault()?.ToArgument().Value) is { } instance
```

Comment:
If I do this
```
private static void Move(List<int> list) { }

public void Go()
{
    var list = new List<int>();
    Move(list);
}
```
this logic will give me the `list` instance, where it shouldn't for the rest of the logic. Because the name will confuse the other conditions.

This needs to be more selective about extensions for the 1st argument. It worked on the old implementation, because it already tested that we're no Enumerable.Count first.

Reply:
I agree that the 'instance' was not the best name for that variable, indeed.
I don't get what you mean with the second suggestion, what was working and now it's not?

Reply:
`instance` name is fine. Problem is that it's getting value when it shouldn't.
For static call, it will take first argument. Even if it isn't an extension. So it will use any random method, and pretend it was extension, or instance call of that method.

It will think that `Move(list)` was actually `list.Move()`.